File: Java/src/test/java/com/gildedrose/GildedRoseTest.java
package com.gildedrose;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

/**
 * Unit tests for the {@link GildedRose} class.
 */
class GildedRoseTest {

    /**
     * Verifies that the item name remains unchanged after updating quality.
     */
    @Test
    void shouldRetainItemNameAfterQualityUpdate() {
        Item[] items = { new Item("foo", 0, 0) };
        GildedRose gildedRose = new GildedRose(items);

        gildedRose.updateQuality();

        assertEquals("fixme", gildedRose.items[0].name);
    }
}

File: csharp.NUnit/GildedRoseTests/GildedRoseTest.cs
using System.Collections.Generic;
using GildedRoseKata;
using NUnit.Framework;

namespace GildedRoseTests
{
    /// <summary>
    /// Unit tests for the <see cref="GildedRose"/> class.
    /// </summary>
    [TestFixture]
    public class GildedRoseTests
    {
        /// <summary>
        /// Verifies that the <see cref="GildedRose.UpdateQuality"/> method updates item properties as expected.
        /// </summary>
        [Test]
        public void UpdateQuality_ShouldUpdateItemNameCorrectly()
        {
            // Arrange
            var items = new List<Item>
            {
                new Item { Name = "foo", SellIn = 0, Quality = 0 }
            };

            var gildedRoseApp = new GildedRose(items);

            // Act
            gildedRoseApp.UpdateQuality();

            // Assert
            Assert.That(items[0].Name, Is.EqualTo("fixme"));
        }
    }
}